### PR TITLE
feat(issue-trace): Replace missing trace alert with a placeholder trace

### LINKED ISSUE #45968: Show a better placeholder when trace cannot be found on an issue
### Background

Events often have a `trace_id` but no associated transaction, which results in a large blue alert message:

Image

This alert is informational, but has some downsides:

- The alert takes up a lot of valuable vertical space, pushing down more useful information
- The alert causes layout shift due to being larger than the loading/success states
- The alert is not actionable, yet draws visual focus away from the actually important information

### Solution

Starting simple, we can display a placeholder that looks similar to the successful trace, but shows this message when you hover over it.

### PR DESCRIPTION
**Pull request by malwilley**
Closes https://github.com/getsentry/sentry/issues/45968

Moves the text inside the alert to a custom placeholder component that will display the help text on hover.

Before:

CleanShot 2023-03-29 at 15 41 13

After:

CleanShot 2023-03-29 at 15 42 05

### CODE REVIEW

File: static/app/views/issueDetails/quickTrace/issueQuickTrace.tsx [hunk 1]
```diff
@@ -1,89 +1,110 @@
...
+  type?: QuickTraceQueryChildrenProps['type'];
+}) {
+  const organization = useOrganization();
+  useEffect(() => {
     trackAdvancedAnalyticsEvent('issue.quick_trace_status', {
       organization,
-      status: quickTrace?.type === 'missing' ? 'transaction missing' : 'trace missing',
-      is_performance_issue: isPerformanceIssue ?? false,
+      status: type === 'missing' ? 'transaction missing' : 'trace missing',
+      is_performance_issue: group.issueCategory === IssueCategory.PERFORMANCE,
```
Comment (scefali) [inline static/app/views/issueDetails/quickTrace/issueQuickTrace.tsx:63]: @malwilley we probably should pass in the `group.issueCategory` as `issue_category` here

File: static/app/views/issueDetails/quickTrace/issueQuickTrace.tsx [hunk 2]
```diff
@@ -1,89 +1,110 @@
...
+      </Tooltip>
+    </QuickTraceWrapper>
+  );
+}
+
+function IssueQuickTrace({group, event, location, organization, quickTrace}: Props) {
+  if (
+    !quickTrace ||
+    quickTrace.error ||
+    quickTrace.trace === null ||
```
Comment (scefali) [inline static/app/views/issueDetails/quickTrace/issueQuickTrace.tsx:98]: @malwilley what about `quickTrace.trace === undefined`? We aren't always good about null vs undefined with serialized data.

Comment (malwilley) [reply static/app/views/issueDetails/quickTrace/issueQuickTrace.tsx:98]: This was the existing logic so it should work fine, but I agree it should be using the `defined` function instead

File: static/app/views/issueDetails/quickTrace/issueQuickTrace.tsx (hunk 1 above)
Comment (malwilley) [reply static/app/views/issueDetails/quickTrace/issueQuickTrace.tsx:63]: Agreed - I think we could actually get rid of this and add `trace_status` to the `issue_details.viewed` event instead? But I'd rather keep the analytics the same here and do that in a separate PR.

Comment (scefali) [reply static/app/views/issueDetails/quickTrace/issueQuickTrace.tsx:63]: @malwilley up to you, i think it's fine to throw it in here as well

**Review by scefali — APPROVED** [review]

**Pull request merged**